docs/USER_GUIDE.md

```diff
@@ -0,0 +1,506 @@
+# GoVPP user guide
+
+* [Binary API generator](#binary-api-generator)
+    * [Installation](#Installation)
+        * [Using the Go toolchain](#using-the--go-toolchain)
+        * [From the source](#from-the-source)
+    * [Plugins](#plugins)
+    * [Options](#options)
+* [VPP API calls](#vpp-api-calls)
+    * [Connection](#connection)
+        * [Synchronous](#synchronous-connect)
+        * [Asynchronous](#asynchronous-connect)
+    * [Sending API messages](#sending-api-messages)
+        * [Channel](#channel)
+        * [Stream client](#stream-client)
+* [The RPC service](#the-rpc-service)
+* [VPP stats](#vpp-stats)
+    * [Low-level API connection](#low-level-stats-api-connection)
+    * [Low-level API usage](#low-level-stats-api-usage)
+
+## Binary API generator
+
+The binary API generator's purpose is to create Go bindings out of the VPP `.json` binary API definitions.
+
+### Installation
+
+Requires Go 1.18 or higher ([download](https://golang.org/dl))
+
+#### Using the  Go toolchain
+
+Latest version (most recent tag):
+
+```
+go install go.fd.io/govpp/cmd/binapi-generator@latest
+```
+
+Development version (master branch):
+
+```
+go install go.fd.io/govpp/cmd/binapi-generator@master
+```
+
+#### From the source
+
+Clone the repository:
+
+```
+git clone https://github.com/FDio/govpp
+```
+
+Call the following make taget inside:
+
+```
+make install-generator
+```
```

Comment:
I think we can omit this section for the User Guide. The previous method is essentially installing from source, just cloning & building in the background.